zapetw/core_windows.go

```diff
@@ -9,48 +9,45 @@ import (
 // <product_name>-<component_name>
 const providername = "ACN-Monitoring"
 
-type ETWCore struct {
+type Core struct {
 	provider  *etw.Provider
 	eventName string
 	encoder   zapcore.Encoder
-	fields    []zapcore.Field
 	zapcore.LevelEnabler
 }
 
-func NewETWCore(eventName string, encoder zapcore.Encoder, levelEnabler zapcore.LevelEnabler) (*ETWCore, error) {
-	provider, err := etw.NewProviderWithOptions(providername)
+func New(providerName, eventName string, encoder zapcore.Encoder, levelEnabler zapcore.LevelEnabler) (zapcore.Core, func(), error) {
+	provider, err := etw.NewProviderWithOptions(providerName)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to create ETW provider")
+		return nil, func() { _ = provider.Close }, errors.Wrap(err, "failed to create ETW provider")
 	}
-	return &ETWCore{
+	return &Core{
 		provider:     provider,
 		eventName:    eventName,
 		encoder:      encoder,
 		LevelEnabler: levelEnabler,
-	}, nil
+	}, func() { _ = provider.Close }, nil
```

Comment:
Did you mean to call `provider.Close()`?

Reply:
yep